packages/run-protocol/src/feeDistributor.js

```diff
@@ -0,0 +1,302 @@
+      [...sourceToCollector.values()].map(collector =>
+        E(collector).collectFees().then(distributeFees),
+      ),
+    );
+  };
+
+  const publicFacet = Far('feeDistributor publicFacet', {
+    distributeFees,
+  });
+
+  const zoe = zcf.getZoeService();
+  const creatorFacet = Far('feeDistributor creatorFacet', {
+    addFeeSource: async sourceP => {
+      const [source, collector] = await Promise.all([
+        sourceP,
+        makeFeeCollector(zoe, sourceP),
+      ]);
+      sourceToCollector.init(source, collector);
+      if (shareConfig.totalShares <= 0n) {
+        return;
+      }
+      // Run once immediately for this source.
+      await E(collector).collectFees().then(distributeFees);
+    },
+    deleteFeeSource: async sourceP => {
+      const source = await sourceP;
+      sourceToCollector.delete(source);
+    },
+    /**
+     * @param {ERef<DepositFacet>} depositFacet
+     */
+    makeDepositFacetDestination: depositFacet => {
+      return Far(`DepositFacetDestination`, {
+        pushPayment: async (payment, _issuer) => {
+          return E(depositFacet).receive(payment);
+        },
+      });
+    },
+    /**
+     * Create a destination that generates invitations and makes Zoe offers.
+     *
+     * @param {string} keyword
+     * @param {any} target
+     * @param {PropertyKey} makeInvitationMethod
+     * @param {any[]} [args]
+     */
+    makeOfferDestination: (
+      keyword,
+      target,
+      makeInvitationMethod,
+      args = [],
+    ) => {
+      return Far(`${String(makeInvitationMethod)} OfferDestination`, {
+        pushPayment: async (payment, issuer) => {
+          const paymentAmount = await E(issuer).getAmountOf(payment);
+
+          // Give the payment to the contract via its invitation.
+          const invitation = E(target)[makeInvitationMethod](...args);
+          const result = E(zoe).offer(
+            invitation,
+            {
+              give: {
+                [keyword]: paymentAmount,
+              },
+            },
+            {
+              [keyword]: payment,
+            },
+          );
+
+          // Assert that the offer completed.
+          await E(result).getOfferResult();
+
+          // We deliberately drop our payouts on the floor, since the ERTP purse
+          // recovery mechanism can get them back to the distributor contract.
+          return paymentAmount;
+        },
+      });
+    },
+
+    setDestinations: async destinations => {
+      shareConfig = makeShareConfig(destinations, rest);
+      // Run once immediately for these destinations.
+      await schedulePayments();
+    },
+  });
+
+  // Start processing collections.
+  startDistributing(schedulePayments, timerService, collectionInterval);
+
+  return harden({
+    creatorFacet,
+    publicFacet,
+  });
+};
+
+/** @typedef {ReturnType<typeof start>['creatorFacet']} FeeDistributorCreatorFacet */
+/** @typedef {ReturnType<typeof start>['publicFacet']} FeeDistributorPublicFacet */
```

Comment:
```suggestion
/** @typedef {Awaited<ReturnType<typeof start>>['creatorFacet']} FeeDistributorCreatorFacet */
/** @typedef {Awaited<ReturnType<typeof start>>['publicFacet']} FeeDistributorPublicFacet */
```
the reason this didn't error before is the `ReturnType<typeof start>` is `any` due to `ContractStartFn`